### PR TITLE
Add java servlet to randomly generate fact

### LINKED ISSUE #5: Add server side quote generator
Use a java servlet and the JS fetch API to randomly generate a quote on the random page.

### PR DESCRIPTION
**Pull request by CarlosM10011**
This pull request adds the server side code for the random fact generator.

Partially addresses issue #5.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/RandomFactServlet.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that generates a random quote. */
+@WebServlet("/random-fact")
+public class RandomFactServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("text/html;");
```
Comment (tinagrasso) [inline portfolio/src/main/java/com/google/sps/servlets/RandomFactServlet.java]: Make the string a class level constant.

```
private const String CONTENT_TYPE = "text/html";
```

File: portfolio/src/main/java/com/google/sps/servlets/RandomFactServlet.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    String[] facts = new String[3];
+    facts[0] = "Our air is composed of mostly nitrogen.";
+    facts[1] = "Moore's Law is an observation.";
+    facts[2] = "The Summit supercomputer can reach up to 200 petaFLOPS.";
```
Comment (tinagrasso) [inline portfolio/src/main/java/com/google/sps/servlets/RandomFactServlet.java]: Prefer a List and make it an instance variabel.

```
private List<String> randomFacts = Lists.of("....", "....");
```

https://docs.oracle.com/javase/9/core/creating-immutable-lists-sets-and-maps.htm#JSCOR-GUID-202D195E-6E18-41F6-90C0-7423B2C9B381

File: portfolio/src/main/java/com/google/sps/servlets/RandomFactServlet.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    Random rand = new Random();
```
Comment (tinagrasso) [inline portfolio/src/main/java/com/google/sps/servlets/RandomFactServlet.java]: Make this an instance variable.

**Review by tinagrasso — CHANGES_REQUESTED** [review]

**Review by hzimz — APPROVED** [review]
LGTM once you address Tina's comments.

File: portfolio/src/main/java/com/google/sps/servlets/RandomFactServlet.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+public class RandomFactServlet extends HttpServlet {
+
+  private final String CONTENT_TYPE = "text/html;";
+  private List<String> facts =
+      Arrays.asList(
+          new String[] {
+            "Our air is composed of mostly nitrogen.",
+            "Moore's Law is an observation.",
+            "The Summit supercomputer can reach up to 200 petaFLOPS."
+          });
```
Comment (tinagrasso) [inline portfolio/src/main/java/com/google/sps/servlets/RandomFactServlet.java]: Remove object arrays and make immutable.

```
private final List<String> facts = Collections.unmodifiableList(Arrays.asList("a", "b", "c"));
```

**Review by tinagrasso — CHANGES_REQUESTED** [review]

File: portfolio/src/main/java/com/google/sps/servlets/RandomFactServlet.java [hunk 5]
```diff
@@ -28,13 +29,14 @@
...
-      Arrays.asList(
-          new String[] {
-            "Our air is composed of mostly nitrogen.",
-            "Moore's Law is an observation.",
-            "The Summit supercomputer can reach up to 200 petaFLOPS."
-          });
+  private final List<String> facts =
+      Collections.unmodifiableList(
+          Arrays.asList(
+              new String[] {
```
Comment (tinagrasso) [inline portfolio/src/main/java/com/google/sps/servlets/RandomFactServlet.java]: Remove the object array. Please look at the javadoc for this method.
https://docs.oracle.com/javase/8/docs/api/index.html?java/util/List.html

**Review by tinagrasso — CHANGES_REQUESTED** [review]

**Review by tinagrasso — APPROVED** [review]

**Pull request merged**